ts/DataGrid/DataGrid.ts

```diff
@@ -956,6 +979,115 @@ class DataGrid {
             this.headerContainer,
             this.outerContainer
         );
+
+        this.updateColumnHeaders();
+    }
+
+
+    /**
+     * Refresh container elements to adapt them to new container dimensions.
+     * @internal
+     */
+    private updateGridElements(): void {
+        this.updateColumnHeaders();
+        this.redrawRowElements();
+        this.updateDragHandlesPosition();
+    }
+
+    /**
+     * Update the column headers of the table.
+     * @internal
+     */
+    private updateColumnHeaders(): void {
+        const headersContainer = this.columnHeadersContainer;
+        if (!headersContainer) {
+            return;
+        }
+
+        // Handle overflowing text in headers.
+        this.columnNames.forEach((columnName, i): void => {
```

Comment:
We avoid using `forEach` in the dashboards for performance reasons. Could you use the basic `for` loop?